packages/react-native-bottom-tabs/android/src/main/java/com/rcttabview/RCTTabView.kt

```diff
@@ -458,6 +462,7 @@ class ReactBottomNavigationView(context: Context) : LinearLayout(context) {
 
   override fun onDetachedFromWindow() {
     super.onDetachedFromWindow()
-    imageLoader.shutdown()
+    imageLoader?.shutdown()
+    imageLoader = null
```

Comment:
Ah! So `onDetachedFromWindow` was called when we opened the modal? I'm thinking whether this is a correct place to shut it down then... 


Maybe we should move the shutdown code somewhere else? Like into an LifecycleObserver: https://developer.android.com/reference/androidx/lifecycle/LifecycleObserver

Reply:
Yeah `onDetachedFromWindow` was called when the modal was opened.

Yep my initial thought was to shutdown the `ImageLoader` somewhere else in the view lifecycle. But my knowledge about Android is limited and I couldn't find a more proper lifecycle method.

LifeCycleObserver looks interesting! But I can't find any example on how to implement it on a View. Can you point me in any direction?

Reply:
tried this code in `init` but doesn't seem to work
```kotlin
    if (context is LifecycleOwner) {
      context.lifecycle.addObserver(object : DefaultLifecycleObserver {
        override fun onDestroy(owner: LifecycleOwner) {
          imageLoader.shutdown()
        }
      })
    }
```

Reply:
@johankasperi Looks like we could implement `onDropViewInstance` from React Native in the ViewManager class: 

https://github.com/facebook/react-native/blob/38190f530ffebd99ae3f70f86bda9e18f15b80e7/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/modal/ReactModalHostManager.kt#L35

We could deinitialize imageLoader there

Reply:
Yeah thats right! Forgot about that one. Will give it a go asap

Reply:
Implemented it now with `onDropViewInstance` instead